### PR TITLE
fixed float precision error on conversion factor and added equivalent_units method

### PR DESCRIPTION
**Pull request by MauriceHendrix**
As I'm currently doing a lot of
if model._get_conversion_factor (..) != 1.0 to avoid generating 1.0* expr

### CODE REVIEW

File: cellmlmanip/units.py [hunk 1]
```diff
@@ -368,13 +368,34 @@ def get_conversion_factor(self,
...
+    def equivalent_units(self, unit_or_expr1, unit_or_expr2):
+        """Returns whether 2 expressions or units are in the same units.
+
+        :param expr_or_unit1: A unit or an expression to look at the units for
+        :param expr_or_unit2: A unit or an expression to look at the units for
+
+        :return: are both parameters in equivalent units (conversion factor == 1.0)
+        """
+        try:
+            print(isinstance(unit_or_expr1, sympy.Expr))
```
Comment (jonc125) [inline cellmlmanip/units.py]: Get rid of this line :)

File: tests/test_unit_conversion.py [hunk 2]
```diff
@@ -1,7 +1,5 @@
 import os
-
```
Comment (jonc125) [inline tests/test_unit_conversion.py]: I don't think isort will like this!

File: tests/test_unit_conversion.py [hunk 3]
```diff
@@ -89,3 +87,18 @@ def test_conversion_factor_same_units(simple_model):
     assert simple_model.units.get_conversion_factor(to_unit=to_unit, from_unit=from_unit) == 1
     # expression to unit
     assert simple_model.units.get_conversion_factor(to_unit=to_unit, expression=expression) == 1
+
+
+def test_equivalent_units(model):
+    time_var = model.get_symbol_by_ontology_term(OXMETA, 'time')
+
+    assert model.units.equivalent_units(time_var, model.units.ureg.ms)
+    assert model.units.equivalent_units(time_var, model.units.ureg.ms)
```
Comment (jonc125) [inline tests/test_unit_conversion.py]: Why is this line repeated?

File: cellmlmanip/units.py [hunk 4]
```diff
@@ -368,13 +368,34 @@ def get_conversion_factor(self,
...
             assert isinstance(expression, sympy.Expr), 'expression must be of type Sympy expression'
-            return self.convert_to(1 * self.summarise_units(expression), to_unit).magnitude
+            conversion_factor = self.convert_to(1 * self.summarise_units(expression), to_unit).magnitude
+
+        if math.isclose(conversion_factor, 1.0):
+            return 1.0
+        else:
+            return conversion_factor
+
+    def equivalent_units(self, unit_or_expr1, unit_or_expr2):
```
Comment (MichaelClerx) [inline cellmlmanip/units.py]: Is this the same as `is_unit_equal` ?

File: tests/test_unit_conversion.py [hunk 5]
```diff
@@ -89,3 +87,18 @@ def test_conversion_factor_same_units(simple_model):
...
+    assert model.units.equivalent_units(model.units.ureg.ms, time_var)
+
+    assert not model.units.equivalent_units(model.units.ureg.ms, model.units.ureg.second)
```
Comment (jonc125) [inline tests/test_unit_conversion.py]: Aren't the pure units cases tested below? There's definitely some test duplication here!

File: cellmlmanip/units.py (hunk 4 above)
Comment (MauriceHendrix) [reply cellmlmanip/units.py]: Actually yes I didn't notice, let me remove my new one.

File: tests/test_unit_conversion.py (hunk 2 above)
Comment (MauriceHendrix) [reply tests/test_unit_conversion.py]: it's not something I changed but we'll see

**Pull request merged**